### PR TITLE
solved?

### PR DESCRIPTION
**Pull request by AlexBondor**
I have managed to obtain the result as expected. I don't know if you were thinking of the approach I've taken. Is it too dirty?

### CODE REVIEW

File: netrc.go [hunk 1]
```diff
@@ -113,11 +122,15 @@ func (netrc Netrc) Save() error {
 
 	writer := bufio.NewWriter(file)
 
+
```
Comment (lorenzleutgeb) [inline netrc.go]: Pointless newline.

File: netrc.go [hunk 2]
```diff
@@ -28,7 +31,7 @@ func Location() string {
 	return os.ExpandEnv("$HOME") + string(os.PathSeparator) + location
 }
 
-func checkPermissions(fileName string) error {
+func CheckPermissions(fileName string) error {
```
Comment (lorenzleutgeb) [inline netrc.go]: Why export this?

**Comment by lorenzleutgeb** [comment]
Consider the following scenario:
- I have an entry for `api.github.com` in my `.netrc`
- This implementation parses `.netrc`, so the entry is added to `Netrc.Entries`.
- Some code changes the entry for `api.github.com` and calls `Save()` on it's instance.

_Changes_ to `Netrc.Entries` will not be reflected. Only new entries are appended (`O_APPEND`).

This is too much of a restriction. Assume client code wants to read an access token and refresh it at some point. Or delete an entry.

Also, your commit is not associated with your GitHub account.

File: netrc.go (hunk 2 above)
Comment (AlexBondor) [inline netrc.go]: My bad! I thought it didn't mean anything special. Now I know:)

**Comment by AlexBondor** [comment]
All right! Thanks for the feedback and updates.

**Comment by lorenzleutgeb** [comment]
Also, "solved?" is not a valid commit message. Nobody will ever have a chance to deduce what changes  this commit introduces.

Apart from that, you have nowhere even tried to detail how your solution works. Neither in the commit message, nor in the pull request description. These places are there for you to describe what you did and why.

> I have managed to obtain the result as expected.

What result? I know, because I sent you an e-mail, but we're on GitHub now.

**Comment by AlexBondor** [comment]
I'm going to take that into account as well.

**Comment by AlexBondor** [comment]
Hello again,

I have managed to save the inital formatting of the entries but upon saving
the entries get mingled. I still need to fix that. I only commited the code
on my branch so I didn't do any pull request yet. I will try to find a
solution for the remaining problem tomorrow.

File: netrc.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package netrc
+package main
```
Comment (lorenzleutgeb) [inline netrc.go:1]: Unacceptable. This is not a program that runs on it's own, it is a library.

File: netrc.go [hunk 4]
```diff
@@ -114,26 +138,55 @@ func (netrc Netrc) Save() error {
...
 
 	writer.Flush()
 
 	return nil
 }
+
+func main() {
+	var netrc, _ = Parse()
+
+	var entry *Entry = nil
```
Comment (lorenzleutgeb) [inline netrc.go:185]: Should be:

``` go
entry := new(Entry)
```

Comment (AlexBondor) [inline netrc.go:185]: About package main..i did that just for quick testing and as i still need
to do the other problem i left it as is.

**Pull request closed without merging**